app/services/district_report_service.rb

```diff
@@ -0,0 +1,90 @@
+class DistrictReportService
+  def initialize(facilities:, selected_date:)
+    @facilities = Array(facilities)
+    # TODO will we ever have multiple facilities? Probably for the District wide reports that
+    # encompass FacilityGroups?
+    @facility = @facilities.first
+    @selected_date = selected_date
+    @data = {
+      controlled_patients: {},
+      registrations: {},
+      cumulative_registrations: 0,
+      quarterly_registrations: []
+    }.with_indifferent_access
+  end
+
+  attr_reader :selected_date, :facilities, :data
+
+  def call
+    compile_control_and_registration_data
+
+    compile_cohort_trend_data
+
+    data
+  end
+
+  def compile_control_and_registration_data
+    -11.upto(0).each do |n|
+      time = selected_date.advance(months: n)
+      formatted_period = time.strftime("%b %Y")
+      key = [@facility.id, time.year.to_s, time.month.to_s]
+
+      period = {cohort_period: :month,
+                registration_month: time.month,
+                registration_year: time.year}
+      monthly_bps = MyFacilities::BloodPressureControlQuery.new(facilities: @facilities, cohort_period: period)
+
+      p [formatted_period, monthly_bps.cohort_controlled_bps.count]
+      controlled_bps = Integer(monthly_bps.cohort_controlled_bps.count || 0)
```

Comment:
Does it look like I'm calling this query correctly here to get the previous 12 months of controlled patient counts? I was debugging this for quite a bit on Wednesday, and the only data I got back in my spec was for two months prior from the current date in the spec. I was not able to get any counts returned for months further back from that, even though I'm iterating through previous months and re-running the query for each month. 

I may be missing something obvious -- any ideas?

Reply:
@rsanheim The bp control query returns bps recorded in the next two months for patients registered in the `:registration_month`. https://github.com/simpledotorg/simple-server/blob/de0ff1dfb830d00c9a7dad11038d5cfd3afb2d9d/app/queries/my_facilities/blood_pressure_control_query.rb#L111

I tried
```
Timecop.freeze(5.months.ago) do
      other_patients = create_list(:patient, 2, recorded_at: Time.current, registration_facility: facility, registration_user: user)
      other_patients.map do |patient|
        create(:blood_pressure, :under_control, facility: facility, patient: patient, recorded_at: Time.current + 1.month)
      end
    end
```
and got

```
["Jul 2019", 0]
0
["Aug 2019", 0]
0
["Sep 2019", 0]
0
["Oct 2019", 0]
0
["Nov 2019", 0]
0
["Dec 2019", 0]
0
["Jan 2020", 2]
2
["Feb 2020", 0]
0
["Mar 2020", 0]
0
["Apr 2020", 2]
2
["May 2020", 0]
0
["Jun 2020", 0]
0
```
Is that what we expect?

Reply:
Ohhh I think that was it. So the BP `recorded_at` needs to be _after_ the patient `recorded_at`? What does the patient `recorded_at` mean ?

Reply:
patient `recorded_at` represents when the patient was first registered in the app.
It's expected that this date would be earlier or the same as when BPs are recorded. When backdated BPs are recorded, we also backdate the patient `recorded_at` if necessary.

Reply:
[C1.1](https://docs.google.com/document/d/1Mo83rTwR_tuuUNnJnWKWlZ49nTzlLxqlrbKYnDj1kLQ/edit#heading=h.acbewxs3lz9z) and C2.1 in this document might help clear some definitions out as well.